pkitesting/src/main/java/io/netty/pkitesting/X509Bundle.java

```diff
@@ -0,0 +1,349 @@
+    }
+
+    /**
+     * Query if this bundle is for a certificate authority root certificate.
+     * @return {@code true} if the {@linkplain #getCertificate() leaf certificate} is a certificate authority,
+     * otherwise {@code false}.
+     */
+    public boolean isCertificateAuthority() {
+        return certPath[0].getBasicConstraints() != -1;
+    }
+
+    /**
+     * Query if this bundle is for a self-signed certificate.
+     * @return {@code true} if the {@linkplain #getCertificate() leaf certificate} is self-signed.
+     */
+    public boolean isSelfSigned() {
+        X509Certificate leaf = certPath[0];
+        return certPath.length == 1 &&
+                leaf.getSubjectX500Principal().equals(leaf.getIssuerX500Principal()) &&
+                Arrays.equals(leaf.getSubjectUniqueID(), leaf.getIssuerUniqueID());
+    }
+
+    /**
+     * Create a {@link TrustManager} instance that trusts the root certificate in this bundle.
+     * @return The new {@link TrustManager}.
+     */
+    public TrustManager toTrustManager() {
+        TrustManagerFactory tmf = toTrustManagerFactory();
+        return tmf.getTrustManagers()[0];
+    }
+
+    /**
+     * Create  {@link TrustManagerFactory} instance that trusts the root certificate in this bundle.
+     * @return The new {@link TrustManagerFactory}.
+     */
+    public TrustManagerFactory toTrustManagerFactory() {
+        try {
+            TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            tmf.init(toKeyStore(EmptyArrays.EMPTY_CHARS));
+            return tmf;
+        } catch (NoSuchAlgorithmException e) {
+            throw new AssertionError("Default TrustManagerFactory algorithm was not available.", e);
+        } catch (KeyStoreException e) {
+            throw new IllegalStateException("Failed to initialize TrustManagerFactory with KeyStore.", e);
+        }
+    }
+
+    /**
+     * Create a {@link KeyStore} with the contents of this bundle.
+     * The root certificate will be a trusted root in the key store.
+     * If this bundle is not a {@linkplain #isCertificateAuthority() certificate authority},
+     * then the private key and certificate path will also be added to the key store.
+     * @param keyEntryPassword The password used to encrypt the private key entry in the key store.
+     * @return The key store.
+     * @throws KeyStoreException If an error occurred when adding entries to the key store.
+     */
+    public KeyStore toKeyStore(char[] keyEntryPassword) throws KeyStoreException {
+        KeyStore keyStore;
+        try {
+            keyStore = KeyStore.getInstance("PKCS12");
+            keyStore.load(null, null);
+        } catch (KeyStoreException | IOException | NoSuchAlgorithmException | CertificateException e) {
+            throw new AssertionError("Failed to initialize PKCS#12 KeyStore.", e);
```

Comment:
I think we should just let the KeyStoreException be thrown and also I would suggest we either throw a `KeyStoreException` for the others or an `IllegalStateException`